### PR TITLE
Parser refactor

### PR DESCRIPTION
**Pull request by dixonwhitmire**
This PR removes the need for transaction specific partners. The transaction set model class is now loaded as part of the "create_parser" function.

Changes include:
* Removed Eligibility Inquiry Parser
* Updated "create_parser" to load the appropriate transaction set model.
* Added test cases for all of the parser functions.

PR Review Changes:
* Correcting usage of NoReturn type hint.

### CODE REVIEW

File: tests/test_create_parser.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+test_create_parser.py
+
+Tests the parser creation function.
+"""
+import pytest
+from x12.parsing import create_parser
+from x12.models import X12Delimiters
+
+
+@pytest.mark.parametrize(
```
Comment (dixonwhitmire) [inline tests/test_create_parser.py:11]: We will add additional parameters to this test as new transactions are supported.

File: x12/parsing.py [hunk 2]
```diff
@@ -275,15 +277,24 @@ def parse(
             self._context.reset_transaction()
             return model
 
-    def __init__(self, x12_delimiters: X12Delimiters) -> NoReturn:
+    def __init__(
+        self,
+        transaction_model: X12SegmentGroup,
+        loop_parsers: Dict,
+        x12_delimiters: Optional[X12Delimiters] = None,
+    ) -> NoReturn:
```
Comment (ccorley) [inline x12/parsing.py:285]: Should '__init__' have a return type of None? https://stackoverflow.com/questions/46779046/correct-type-annotation-for-init
Looks like NoReturn is reserved for functions that never return or that always throw an exception.

**Review by ccorley — CHANGES_REQUESTED** [review]
Looks great, just one comment.

File: x12/parsing.py (hunk 2 above)
Comment (dixonwhitmire) [reply x12/parsing.py:285]: Hey @ccorley - Thanks! That's a good catch. 
Reference - https://docs.python.org/3/library/typing.html#typing.NoReturn
You know I think I've misused this type hint consistent throughout the project. I'll correct that and will push up changes.

**Review by ccorley — APPROVED** [review]

**Pull request merged**